app/actions/definitions/documents.tsx

```diff
@@ -352,7 +352,18 @@ export const duplicateDocument = createAction({
  * of the collection for all collection members to see.
  */
 export const pinDocumentToCollection = createAction({
-  name: ({ t }) => t("Pin to collection"),
+  name: ({ activeDocumentId = "", t, stores }) => {
+    const selectedDocument = stores.documents.get(activeDocumentId);
+    const selectedDocumentCollectionName = selectedDocument
+      ? stores.documents.getCollectionForDocument(selectedDocument)?.name
+      : "";
+    return selectedDocumentCollectionName
+      ? t("Pin to {{collectionName}}", {
+          collectionName: selectedDocumentCollectionName,
+        })
+      : "";
```

Comment:
```suggestion
    const selectedDocument = stores.documents.get(activeDocumentId);
    const selectedDocumentCollectionName = selectedDocument
      ? stores.documents.getCollectionForDocument(selectedDocument)?.name
      : undefined;
    return selectedDocumentCollectionName
      ? t("Pin to {{collectionName}}", {
          collectionName: selectedDocumentCollectionName,
        })
      : t("Pin to collection");
```

We can fall back to the old language otherwise it's possible to have a menu item with no text